Yubico.YubiKey/src/Yubico/YubiKey/Piv/PivAlgorithmExtensions.cs

```diff
@@ -12,13 +12,168 @@
+                    { Algorithm = PivAlgorithm.EccEd25519, KeyDefinition = KeyDefinitions.Ed25519 } },
+                { PivAlgorithm.EccX25519, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.EccX25519, KeyDefinition = KeyDefinitions.X25519 } },
+                { PivAlgorithm.Aes128, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Aes128, KeyDefinition = KeyDefinitions.AES128 } },
+                { PivAlgorithm.Aes192, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Aes192, KeyDefinition = KeyDefinitions.AES192 } },
+                { PivAlgorithm.Aes256, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Aes256, KeyDefinition = KeyDefinitions.AES256 } },
+                { PivAlgorithm.TripleDes, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.TripleDes, KeyDefinition = KeyDefinitions.TripleDes } },
+                { PivAlgorithm.None, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.None, KeyDefinition = null! } },
+                { PivAlgorithm.Pin, new PivAlgorithmDefinition 
+                    { Algorithm = PivAlgorithm.Pin, KeyDefinition = null! } },
+            };
+
+            // Create reverse mapping (KeyType to PivAlgorithm)
+            _keyTypeToPivAlgorithmMap = new Dictionary<KeyType, PivAlgorithm>
+            {
+                { KeyType.RSA1024, PivAlgorithm.Rsa1024 },
+                { KeyType.RSA2048, PivAlgorithm.Rsa2048 },
+                { KeyType.RSA3072, PivAlgorithm.Rsa3072 },
+                { KeyType.RSA4096, PivAlgorithm.Rsa4096 },
+                { KeyType.P256, PivAlgorithm.EccP256 },
+                { KeyType.P384, PivAlgorithm.EccP384 },
+                { KeyType.P521, PivAlgorithm.EccP521 },
+                { KeyType.Ed25519, PivAlgorithm.EccEd25519 },
+                { KeyType.X25519, PivAlgorithm.EccX25519 },
+                { KeyType.AES128, PivAlgorithm.Aes128 },
+                { KeyType.AES192, PivAlgorithm.Aes192 },
+                { KeyType.AES256, PivAlgorithm.Aes256 },
+                { KeyType.TripleDes, PivAlgorithm.TripleDes },
+            };
+        }
+
+        /// <summary>
+        /// Gets the PivAlgorithmDefinition for a given KeyDefinition.
+        /// </summary>
+        /// <param name="keyDefinition">The key definition to look up.</param>
+        /// <returns>The corresponding PivAlgorithmDefinition.</returns>
+        /// <exception cref="InvalidOperationException">Thrown when no matching algorithm definition is found.</exception>
+        public static PivAlgorithmDefinition GetPivKeyDefinition(this KeyDefinition keyDefinition)
+        {
+            if (!_keyTypeToPivAlgorithmMap.TryGetValue(keyDefinition.KeyType, out var algorithm))
+            {
+                throw new InvalidOperationException($"No PIV algorithm mapping found for key type: {keyDefinition.KeyType}");
```

Comment:
I just realized that in the SDK one throws exceptions in two different ways:
1. passing parameters directly as here
2. using `string.Format()`

What are the rules for each of the approaches?